line-vertical/index.js

```diff
@@ -0,0 +1,349 @@
+                .frameName(frameName)
+                .intraday(intraday);
+
+            currentFrame.plot()
+                .call(myYAxis);
+
+            const yLabelWidth = myYAxis.labelWidth();
+
+
+            myYAxis
+                .tickSize(currentFrame.dimension().width - yLabelWidth);
+
+            myYAxis1
+                .domain(yDomain)
+                .range([0, currentFrame.dimension().height])
+                .align('right')
+                .fullYear(true)
+                .interval(interval)
+                .tickSize(currentFrame.dimension().width - yLabelWidth)
+                .minorAxis(minorAxis)
+                .minorTickSize(currentFrame.rem() * 0.3)
+                .frameName(frameName)
+                .intraday(intraday);
+
+            myYAxis.yLabel().selectAll('.tick').remove();
+
+            // Draw the yAxis first, this will position the yAxis correctly and
+            // measure the width of the label text
+            currentFrame.plot()
+                .call(myYAxis)
+                .call(myYAxis1);
+
+            // return the value in the variable newMargin
+
+            const newMarginR = myYAxis1.labelWidth() + currentFrame.margin().right;
+            currentFrame.margin({ right: newMarginR });
+
+
+            const newMarginL = myYAxis.labelWidth() + currentFrame.margin().left;
+            // Use newMargin redefine the new margin and range of xAxis
+            currentFrame.margin({ left: newMarginL });
+            myYAxis.yLabel().attr('transform', `translate(${(myYAxis.tickSize() - myYAxis.labelWidth())},0)`);
+            myYAxis.yLabel().selectAll('.tick text').attr('dx', -(currentFrame.rem() / 2));
+            myYAxis1.yLabel().selectAll('.tick text').attr('dx', (currentFrame.rem() / 2));
+            myYAxis1.yLabelMinor().attr('transform', `translate(${(myYAxis.tickSize() - myYAxis.labelWidth())},0)`);
+
+            d3.select(currentFrame.plot().node().parentNode)
+                .call(currentFrame);
+
+            // axisHighlight.append("rect")
+            //   .attr("width", currentFrame.dimension().width)
+            //   .attr("height",currentFrame.dimension().height)
+            //   .attr("fill","#ededee");
+
+
+            myXAxis
+                .domain([Math.min(xMin, valueExtent[0]), Math.max(xMax, valueExtent[1])])
+                .range([0, currentFrame.dimension().width])
+                .numTicks(numTicks)
+                .tickSize(tickSize)
+                .xAxisHighlight(xAxisHighlight)
+                .align(xAxisAlign)
+                .frameName(frameName)
+                .invert(invertScale)
+                .logScale(logScale);
+
+
+            // Draw the xAxis
+            currentFrame.plot()
+                .call(myXAxis);
+
+            if (xAxisAlign === 'bottom') {
+                myXAxis.xLabel().attr('transform', `translate(0,${currentFrame.dimension().height})`);
+            }
+            if (xAxisAlign === 'top') {
```

Comment:
Should this be a separate `if` statement, or does it make more sense as an `else` statement?